File: DSA/ShlokMadan_590017851/Day_33/Question1.java
public class Question1 
{
    public static String removeOuterParentheses(String s) 
    {
        StringBuilder ans=new StringBuilder();
        int depth=0;
        for(char ch:s.toCharArray())
            {
            if(ch=='(')
                {
                if(depth>0)
                    ans.append(ch);
                depth++;
            } 
            else 
                {
                depth--;
                if(depth>0)
                    ans.append(ch);
            }
        }
        return ans.toString();
    }
    public static void main(String[] args) 
    {
        String s="(()())(())";
        System.out.println(removeOuterParentheses(s));
    }
}

File: DSA/Pulkit_Singh_590022181/Day33/Question1.c
#include <stdio.h>
#include <string.h>

void removeOuterParentheses(char s[], char result[]) {
    int depth = 0;
    int j = 0;

    for (int i = 0; s[i] != '\0'; i++) {

        if (s[i] == '(') {
            if (depth > 0)
                result[j++] = s[i];

            depth++;
        }
        else {
            depth--;

            if (depth > 0)
                result[j++] = s[i];
        }
    }

    result[j] = '\0';
}

int main() {
    char s[] = "(()())(())";
    char result[1000];

    removeOuterParentheses(s, result);

    printf("Result: %s\n", result);

    return 0;
}